delta/imagery/imagery_dataset.py

```diff
@@ -110,18 +155,35 @@ def _load_tensor_imagery(self, is_labels, image_index, bbox):
         return r
 
     def _tile_images(self):
+        """Return a Dataset generator object which will cycle through all tiles in all input images"""
         max_block_bytes = config.io.block_size_mb() * 1024 * 1024
+
+        # Define a local generator function to be passed into a TF dataset function.
         def tile_generator():
-            tgs = []
-            for i in range(len(self._images)):
 
-                if self._resume_mode:
-                    # TODO: Improve feature to work with multiple epochs
-                    # Skip images which we have already read some number of tiles from
-                    if self._get_image_read_count(self._images[i]) > config.io.resume_cutoff():
-                        continue
+            # Get a list of input image indices in a random order
+            num_images = len(self._images)
+            indices    = list(range(num_images))
+            random.Random(0).shuffle(indices) # Use consistent random ordering
 
+            # Local function to get the tile list for a single image index
+            def get_image_tile_list(i):
                 try:
+
+                    # If we need to skip this file because of the read count, no need to look up tiles.
+                    file_path = self._images[i]
+                    log_path  = self._get_image_read_log_path(file_path)
+                    #print('get_image_tile_list for index ' + str(i) + ' -> ' + file_path)
+                    if log_path:
+                        (need_to_check, count) = self._read_access_count_file(log_path)
+                        if self._resume_mode and need_to_check and (count > config.io.resume_cutoff()):
```

Comment:
We don't need the log_path or file_path unless resume_mode is true, right? I think it will be clearer if we put that if around all this (and remove all the commented out prints)